### ISSUE TITLE
[Bug]: Some RSS feeds seem to cause the application to crash

### ISSUE THREAD

**Issue by jeremiahelroy** [advplyr/audiobookshelf#1996]
### Describe the issue

Certain podcast rss feeds seem to cause the server to crash and go completely down.  This seems to happen both when trying to download episodes and also matching from existing files.  I have tried adding the files and then editing the podcast with the rss feed in order to match and also doing a search to add the podcast from scratch.

### Steps to reproduce the issue

1. Add the `All Systems Goku` podcast from Giantbomb,  The rss feed is https://www.giantbomb.com/podcast-xml/all-systems-goku.
2. Download an episode (alternatively quick match on an existing download)
4. This should crash the server

`metadata/log/daily`
This unfortunately contains no information and just says it's downloading the cover and then ends
```
{"timestamp":"2023-08-06 15:31:32","source":"Logger.js:114","message":"[CoverManager] Downloaded libraryItem cover \"/metadata/items/a0f69b04-45b5-472a-805b-20e1ba095e1c/cover.jpg\" from url \"https://is5-ssl.mzstatic.com/image/thumb/Podcasts125/v4/28/4e/20/284e20fc-dba3-4d41-595c-fe69a669c46c/mza_16604676203343182380.jpg/600x600bb.jpg\" for \"All Systems Goku\"","levelName":"INFO","level":2}
```

`unraid docker log`
This is ephemeral so I cannot grab the complete log and it only contains a partial log, but maybe something can be gleaned from this.
```
            channels: 1,
            channelLayout: 'mono',
            chapters: [],
            embeddedCoverArt: 'mjpeg',
            metaTags: [Object],
            mimeType: 'audio/mpeg'
          },
          chapters: [],
          extraData: {},
          createdAt: 2023-08-01T03:28:40.263Z,
          updatedAt: 2023-08-06T22:38:19.073Z,
          podcastId: '9eac914e-be1d-4a1b-bb35-fc5747f83cee'
        },
        _previousDataValues: {
          id: 'ebd89968-e69e-4449-88e5-9bd7259cd066',
          index: 1,
          season: null,
          episode: null,
          episodeType: null,
          title: 'All Systems Goku 34',
          subtitle: null,
          description: "Hey, hi, thanks for listening to this thing. I don't think we ever thought this would take off the way it did and it's been a lot of fun to do it. And that's a shoot, brother.",
          pubDate: '2019-03-01 15:33',
          enclosureURL: null,
          enclosureSize: null,
          enclosureType: null,
          publishedAt: 2019-03-01T23:33:00.000Z,
          audioFile: {
            index: 1,
            ino: '651614571405104789',
            metadata: [Object],
            addedAt: 1690860520237,
            updatedAt: 1690860520237,
            trackNumFromMeta: null,
            discNumFromMeta: null,
            trackNumFromFilename: null,
            discNumFromFilename: null,
            manuallyVerified: false,
            invalid: false,
            exclude: false,
            error: null,
            format: 'MP2/3 (MPEG audio layer 2/3)',
            duration: 4064.4965,
            bitRate: 64000,
            language: null,
            codec: 'mp3',
            timeBase: '1/14112000',
            channels: 1,
            channelLayout: 'mono',
            chapters: [],
            embeddedCoverArt: 'mjpeg',
            metaTags: [Object],
            mimeType: 'audio/mpeg'
          },
          chapters: [],
          extraData: {},
          createdAt: 2023-08-01T03:28:40.263Z,
          updatedAt: 2023-08-01T03:28:40.263Z,
          podcastId: '9eac914e-be1d-4a1b-bb35-fc5747f83cee'
        },
        uniqno: 1,
        _changed: Set(8) {
          'episodeType',
          'subtitle',
          'description',
          'pubDate',
          'enclosureURL',
          'enclosureSize',
          'enclosureType',
          'publishedAt'
        },
        _options: {
          isNewRecord: false,
          _schema: null,
          _schemaDelimiter: '',
          include: undefined,
          includeNames: undefined,
          includeMap: undefined,
          includeValidated: true,
          raw: true,
          attributes: undefined
        },
        isNewRecord: false
      },
      validatorKey: 'not_a_string',
      validatorName: null,
      validatorArgs: []
    }
  ]
}
```

### Audiobookshelf version

v2.3.3 - advplyr/audiobookshelf:latest - unraid

### How are you running audiobookshelf?

Docker

**Comment by nichwall** [comment]
Is your server set to log Debug? That should help you see the rest of the logs. I was able to reproduce a similar issue over the weekend (from a bug report on Discord, but the user who reported it hasn't responded yet)

I got an error of `id must be unique` after all episodes had been downloaded within ABS, resulting in the most recent download file slowing as incomplete within ABS (but was complete when tested with ffprobe/other players). However, once the podcast existed, I was able to delete the "bad" podcast file, and redownload it within ABS.

Are you trying to add a new podcast, or just adding new episodes/files to a podcast that is already on the server?

**Comment by jeremiahelroy** [comment]
Alright got debug logs.  There's nothing in there that screams out to me, but maybe it'll give you a better idea what's happening.  This is from me doing a quick match on existing files.

Here's some screenshots of what happens to give some more details.

`At home`
![Screenshot 2023-08-07 192126](https://github.com/advplyr/audiobookshelf/assets/2184261/f9c1cace-338a-42db-95ba-2333197483ca)

`At podcast screen`
![Screenshot 2023-08-07 192159](https://github.com/advplyr/audiobookshelf/assets/2184261/611e46a5-90c2-4ff7-a586-048533325d5e)

`Clicking quick-match`
![Screenshot 2023-08-07 192246](https://github.com/advplyr/audiobookshelf/assets/2184261/e51aa833-e414-4a08-bed2-a128f0480d9f)

`Immediately errors out`
![Screenshot 2023-08-07 192306](https://github.com/advplyr/audiobookshelf/assets/2184261/73728524-9556-4572-8529-787613919e6b)

`Server immediately down`
![Screenshot 2023-08-07 192327](https://github.com/advplyr/audiobookshelf/assets/2184261/c04c3ef7-c293-4224-9e80-0c89ef3941b9)

 ```
{"timestamp":"2023-08-07 16:15:01","source":"Logger.js:108","message":"Set Log Level to DEBUG","levelName":"DEBUG","level":1}
{"timestamp":"2023-08-07 16:16:49","source":"Logger.js:108","message":"[podcastUtils] getPodcastFeed for \"https://www.giantbomb.com/podcast-xml/all-systems-goku\" success - parsing xml","levelName":"DEBUG","level":1}
{"timestamp":"2023-08-07 16:16:49","source":"Logger.js:108","message":"[Scanner] quickMatchPodcastEpisodes: Found episode match for \"All Systems Goku 34\" => All Systems Goku 34","levelName":"DEBUG","level":1}
{"timestamp":"2023-08-07 16:16:49","source":"Logger.js:108","message":"[Scanner] quickMatchPodcastEpisodes: Found episode match for \"All Systems Goku 33\" => All Systems Goku 33","levelName":"DEBUG","level":1}
{"timestamp":"2023-08-07 16:16:49","source":"Logger.js:108","message":"[Scanner] quickMatchPodcastEpisodes: Found episode match for \"All Systems Goku 32\" => All Systems Goku 32","levelName":"DEBUG","level":1}
{"timestamp":"2023-08-07 16:16:49","source":"Logger.js:108","message":"[Scanner] quickMatchPodcastEpisodes: Found episode match for \"All Systems Goku 31\" => All Systems Goku 31","levelName":"DEBUG","level":1}
{"timestamp":"2023-08-07 16:16:49","source":"Logger.js:108","message":"[Scanner] quickMatchPodcastEpisodes: Found episode match for \"All Systems Goku 30\" => All Systems Goku 30","levelName":"DEBUG","level":1}
{"timestamp":"2023-08-07 16:16:49","source":"Logger.js:108","message":"[Scanner] quickMatchPodcastEpisodes: Found episode match for \"All Systems Goku 29\" => All Systems Goku 29","levelName":"DEBUG","level":1}
{"timestamp":"2023-08-07 16:16:49","source":"Logger.js:108","message":"[Scanner] quickMatchPodcastEpisodes: Found episode match for \"All Systems Goku 28\" => All Systems Goku 28","levelName":"DEBUG","level":1}
{"timestamp":"2023-08-07 16:16:49","source":"Logger.js:108","message":"[Scanner] quickMatchPodcastEpisodes: Found episode match for \"All Systems Goku 27\" => All Systems Goku 27","levelName":"DEBUG","level":1}
{"timestamp":"2023-08-07 16:16:49","source":"Logger.js:108","message":"[Scanner] quickMatchPodcastEpisodes: Found episode match for \"All Systems Goku 26\" => All Systems Goku 26","levelName":"DEBUG","level":1}
{"timestamp":"2023-08-07 16:16:49","source":"Logger.js:108","message":"[Scanner] quickMatchPodcastEpisodes: Found episode match for \"All Systems Goku 25\" => All Systems Goku 25","levelName":"DEBUG","level":1}
{"timestamp":"2023-08-07 16:16:49","source":"Logger.js:108","message":"[Scanner] quickMatchPodcastEpisodes: Found episode match for \"All Systems Goku 24\" => All Systems Goku 24","levelName":"DEBUG","level":1}
{"timestamp":"2023-08-07 16:16:49","source":"Logger.js:108","message":"[Scanner] quickMatchPodcastEpisodes: Found episode match for \"All Systems Goku 23\" => All Systems Goku 23","levelName":"DEBUG","level":1}
{"timestamp":"2023-08-07 16:16:49","source":"Logger.js:108","message":"[Scanner] quickMatchPodcastEpisodes: Found episode match for \"All Systems Goku 21\" => All Systems Goku 21","levelName":"DEBUG","level":1}
{"timestamp":"2023-08-07 16:16:49","source":"Logger.js:108","message":"[Scanner] quickMatchPodcastEpisodes: Found episode match for \"All Systems Goku Special: History of Trunks\" => All Systems Goku Special: History of Trunks","levelName":"DEBUG","level":1}
{"timestamp":"2023-08-07 16:16:49","source":"Logger.js:108","message":"[Scanner] quickMatchPodcastEpisodes: Found episode match for \"All Systems Goku 20\" => All Systems Goku 20","levelName":"DEBUG","level":1}
{"timestamp":"2023-08-07 16:16:49","source":"Logger.js:108","message":"[Scanner] quickMatchPodcastEpisodes: Found episode match for \"All Systems Goku 19\" => All Systems Goku 19","levelName":"DEBUG","level":1}
{"timestamp":"2023-08-07 16:16:49","source":"Logger.js:108","message":"[Scanner] quickMatchPodcastEpisodes: Found episode match for \"All Systems Goku 18\" => All Systems Goku 18","levelName":"DEBUG","level":1}
{"timestamp":"2023-08-07 16:16:49","source":"Logger.js:108","message":"[Scanner] quickMatchPodcastEpisodes: Found episode match for \"All Systems Goku 17\" => All Systems Goku 17","levelName":"DEBUG","level":1}
{"timestamp":"2023-08-07 16:16:49","source":"Logger.js:108","message":"[Scanner] quickMatchPodcastEpisodes: Found episode match for \"All Systems Goku 16\" => All Systems Goku 16","levelName":"DEBUG","level":1}
{"timestamp":"2023-08-07 16:16:49","source":"Logger.js:108","message":"[Scanner] quickMatchPodcastEpisodes: Found episode match for \"All Systems Goku 15\" => All Systems Goku 15","levelName":"DEBUG","level":1}
{"timestamp":"2023-08-07 16:16:49","source":"Logger.js:108","message":"[Scanner] quickMatchPodcastEpisodes: Found episode match for \"All Systems Goku 14\" => All Systems Goku 14","levelName":"DEBUG","level":1}
{"timestamp":"2023-08-07 16:16:49","source":"Logger.js:108","message":"[Scanner] quickMatchPodcastEpisodes: Found episode match for \"All Systems Goku 13\" => All Systems Goku 13","levelName":"DEBUG","level":1}
{"timestamp":"2023-08-07 16:16:49","source":"Logger.js:108","message":"[Scanner] quickMatchPodcastEpisodes: Found episode match for \"All Systems Goku 12\" => All Systems Goku 12","levelName":"DEBUG","level":1}
{"timestamp":"2023-08-07 16:16:49","source":"Logger.js:108","message":"[Scanner] quickMatchPodcastEpisodes: Found episode match for \"All Systems Goku 11\" => All Systems Goku 11","levelName":"DEBUG","level":1}
{"timestamp":"2023-08-07 16:16:49","source":"Logger.js:108","message":"[Scanner] quickMatchPodcastEpisodes: Found episode match for \"All Systems Goku 10\" => All Systems Goku 10","levelName":"DEBUG","level":1}
{"timestamp":"2023-08-07 16:16:49","source":"Logger.js:108","message":"[Scanner] quickMatchPodcastEpisodes: Found episode match for \"All Systems Goku 22\" => All Systems Goku 22","levelName":"DEBUG","level":1}

**Comment by advplyr** [comment]
I found the issue. Specific to this RSS feed. Will be fixed in the next release

**Comment by advplyr** [comment]
Fixed in [v2.5.0](https://github.com/advplyr/audiobookshelf/releases/tag/v2.5.0)
